### ISSUE TITLE
'typeof(T).FullName' is never null

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#45774]
**Version Used**: 3.7

**Steps to Reproduce**:

```csharp
#nullable enable

class C
{
  string TypeName => typeof(C).FullName;
}
```

**Expected Behavior**:

No warning is reported for the expression returned from `TypeName`.

**Actual Behavior**:

> warning CS8603: Possible null reference return.

**Comment by tannergooding** [comment]
There are lots of APIs in .NET, especially in the reflection/metadata namespaces that will never return `null` except for some weird edge case (generally around dynamically created types).

Is this one really that special in comparison?

**Comment by jaredpar** [comment]
Moving to csharplang as this is really a language design topic. Essentially the question comes down to whether or not we want the language to override the annotations of the runtime library in specific contexts like `typeof`.
